cargo-miri/bin.rs

```diff
@@ -949,7 +949,10 @@ fn phase_runner(binary: &Path, binary_args: env::Args, phase: RunnerPhase) {
             // to emit the diagnostic structure that Cargo would consume from rustc to emit colored
             // diagnostics, and ask rustc to emit them.
             // See https://github.com/rust-lang/miri/issues/2037
-            if arg.split(',').any(|a| a == "diagnostic-rendered-ansi") {
+            // First skip over the leading `=`, then check for diagnostic-rendered-ansi in the
+            // comma-separated list
+            if suffix.strip_prefix('=').unwrap().split(',').any(|a| a == "diagnostic-rendered-ansi")
```

Comment:
This makes the `assert!(suffix.starts_with('='));` above redundant.

I think it's best to put the `let suffix = suffix.strip_prefix('=').unwrap()` where the `assert` is, also to keep this line a bit shorter.

Reply:
This comment is still open.